js/interface.js

```diff
@@ -1193,6 +1193,38 @@ var DynamicLists = (function() {
       _this.handleTokensSelection();
       _this.loadTokenFields();
     },
+    sortDataSourcesListByCurrentAppId: function(dataSources) {
+      var groupedDataSources = [
+        {
+          id: 'currentAppDSs',
+          text: 'This app',
+          name: 'currentApp',
+          children: []
+        },
+        {
+          id: 'OtherDSs',
+          text: 'Other apps',
+          name: 'otherApp',
+          children: []
+        }
+      ];
+
+      dataSources.forEach(function(dataSource) {
+        if (dataSource.appId === appId) {
```

Comment:
@YaroslavOvdii This suggests that we're using `dataSource.appId` to determine if the data source is associated with an app.

In the following scenario:

1. A data source is created, but not for the app
1. The data source is associated with the app using the developer option feature in Studio

In this case, is the data source also added to the `This app` list?

Reply:
@tonytlwu 

> A data source is created, but not for the app

If the data source doesn't have an `appId` or `apps` array doesn't contain current appId.
This data source will be added to the `Other apps` list.

> The data source is associated with the app using the developer option feature in Studio

And if a situation will be opposite to described above we will add this data source to the `This app` list.